seccomp/umount.go

```diff
@@ -113,19 +120,221 @@ func (u *umountSyscallInfo) process() (*sysResponse, error) {
 		return u.tracer.createSuccessResponse(u.reqId), nil
 	}
 
+	// Verify if the umount op is addressing an immutable resource and prevent
+	// it if that's the case.
+	if ok, resp := u.umountAllowed(mip); ok == false {
+		return resp, nil
+	}
+
 	// Not a mount we manage, have the kernel do the unmount.
 	return u.tracer.createContinueResponse(u.reqId), nil
 }
 
+// umountAllowed purpose is to prevent immutable resources from being
+// unmounted.
+//
+// Method will return 'true' when the unmount operation is deemed legit, and
+// will return 'false' otherwise.
+func (u *umountSyscallInfo) umountAllowed(
+	mip domain.MountInfoParserIface) (bool, *sysResponse) {
+
+	// Skip file-systems explicitly handled by sysbox-fs.
+	if u.FsType == "proc" || u.FsType == "sysfs" {
+		return true, nil
+	}
+
+	// There must be mountinfo state present for this target. Otherwise, let
+	// kernel handle the error.
+	info := mip.GetInfo(u.Target)
+	if info == nil {
+		return false, u.tracer.createContinueResponse(u.reqId)
+	}
```

Comment:
Should we return error here, instead of letting the kernel handle it?

Reply:
It's better for us to return the error, otherwise we open a TOCTOU error: by the time the kernel checks, who's to say that the mountpoint won't be there?